sources/packages/backend/libs/sims-db/src/entities/application-restriction-bypass.model.ts

```diff
@@ -0,0 +1,134 @@
+import {
+  Column,
+  Entity,
+  JoinColumn,
+  ManyToOne,
+  OneToOne,
+  PrimaryGeneratedColumn,
+} from "typeorm";
+import { ColumnNames, TableNames } from "../constant";
+import { RecordDataModel } from "./record.model";
+import {
+  User,
+  Note,
+  StudentRestriction,
+  RestrictionBypassBehaviors,
+  Application,
+} from "@sims/sims-db";
+
+/**
+ * Restrictions bypass that allow awards to be disbursed ignoring
+ * restrictions at the student application level.
+ */
+@Entity({ name: TableNames.ApplicationRestrictionBypasses })
+export class ApplicationRestrictionBypass extends RecordDataModel {
+  /**
+   * Auto-generated sequential primary key column.
+   */
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  /**
+   * Reference to the student application that will have the bypass applied.
+   */
+  @ManyToOne(() => Application, {
+    eager: false,
+    cascade: false,
+  })
+  @JoinColumn({
+    name: "application_id",
+    referencedColumnName: ColumnNames.ID,
+  })
+  application: Application;
+
+  /**
+   * Active student restriction to be bypassed.
+   */
+  @ManyToOne(() => StudentRestriction, {
```

Comment:
I am having some confusion understanding this relation.

Reply:
Thank You for explaining on the call.

Reply:
Assuming the Student has tow active restrictions (e.g. `APPC`, `B6A`) and three applications.

The bypasses can be added as below.

| Application| Restriction | Active?|
| -------- | ------- |------- |
| 1111| APPC |true|
| 1111| B6A|true|
| 2222| B6A|true|
| 3333| B6A|false|

In the above example:
- application `1111` bypasses both restrictions;
- application `2222` bypasses only `B6A`.
- application `3333` had a bypass for `B6A` but it is no longer active, meaning the application will have the next disbursement, if any, blocked.